src/bot_methods.js

```diff
@@ -1,314 +1,498 @@
+    );
+
+    sendWordOfTheDay(message, true);
+
+    interval = bot.setInterval(sendWordOfTheDay, MSEC_PER_DAY, message, true);
+  } else {
+    console.log("Bot did not start, user was not a moderator or admin");
+    //logger.logMessage(message);
+
+    message.channel.send(
+      "Bot did not start, user must be a moderator or admin to use this command."
+    );
+  }
+}
+
+/**
+ * Handles the "stop" command
+ * stops automatic sending wotd messages
+ * only an admin or moderator can use this command.
+ * @param {Message} message
+ */
+function handleCommandStop(message) {
+  if (checkIfAdmin(message)) {
+    console.log("Bot stopped!");
+    //logger.logMessage(message);
+
+    message.channel.send(
+      "Bot stopped! New words will not be posted automatically."
+    );
+
+    bot.clearInterval(interval);
+  } else {
+    console.log("Bot did not stop, user was not a moderator or admin");
+    //logger.logMessage(message);
+
+    message.channel.send(
+      "Bot did not stop, user must be a moderator or admin to use this command."
+    );
+  }
+}
+
+/**
+ * Handles the "help" command
+ * displays information about the bot commands in chat
+ * @param {Message} message
+ */
+function handleCommandHelp(message) {
+  console.log("Bot helped!");
+
+  let help =
+    "Commands for the word of the day bot:\n" +
+    "```" +
+    "$word x  - Searches word x in the dictionary and posts it if available\n" +
+    "$start - Starts automatic sending the word of the day - once a day, from current time (requires moderator permissions)\n" +
+    "$stop  - Stops automatic sending the word of the day (requires moderator permissions)\n" +
+    "$goodbot - Shows your appreciation for the bot\n" +
+    "$goodboy - Woof!\n" +
+    "```";
+
+  message.channel.send(help);
+}
+
+function handleChallengeMode(message) {
+  let newWord = challengeMode(message);
+  console.log(newWord[0]);
+  console.log(newWord[1]);
+  const filter = response => response.content.toLowerCase() == newWord[1].toLowerCase() || response.content.toLowerCase() == 'stop' || response.content.toLowerCase() == 'next';
+  message.channel.send("Translate: " + newWord[0]).then(() => {
+    message.channel.awaitMessages(filter, { maxMatches: 1, time: 15000, errors: ['time'] })
+      .then(collected => {
+        console.log(collected.first().content);
+        if (collected.first().content.toLowerCase() == 'stop') {
+          message.channel.send('All done!');
+          return;
+        } else if (collected.first().content.toLowerCase() == 'next') {
+          message.channel.send('The answer is: **' + newWord[1] + '**');
+          handleChallengeMode(message);
+        } else {
+          message.channel.send(`${collected.first().author} got the correct answer!`);
+          handleChallengeMode(message);
+        }
+      })
+      .catch(collected => {
+        message.channel.send('The answer I have is: **' + newWord[1] + "**");

```

Comment:
do we enter this catch block after some time has passed without a message? might add a comment here